cocotb/decorators.py

```diff
@@ -190,24 +178,18 @@ def _advance(self, outcome):
             outcome: The :any:`outcomes.Outcome` object to resume with.
 
         Returns:
-            The object yielded from the coroutine
+            The object yielded from the coroutine or None if coroutine finished
 
-        Raises:
-            CoroutineComplete: If the coroutine returns or throws an error, self._outcome is set, and
-           :exc:`CoroutineComplete` is thrown.
         """
         try:
             self._started = True
             return outcome.send(self._coro)
         except ReturnValue as e:
             self._outcome = outcomes.Value(e.retval)
-            raise CoroutineComplete()
         except StopIteration as e:
             self._outcome = outcomes.Value(e.value)
-            raise CoroutineComplete()
         except BaseException as e:
```

Comment:
I don't understand why this is `BaseException` and not `Exception`.

Reply:
@eric-wieser had changed this from `Exception` in https://github.com/cocotb/cocotb/commit/b514cad8b083e07acfd68340f53c5d57f2352a08, maybe he remembers why?

Reply:
The idea was to catch Ctrl+C so that it is propagated to the calling coroutine.